google/cloud/ndb/_cache.py

```diff
@@ -304,7 +304,33 @@ class _GlobalCacheSetBatch(_GlobalCacheBatch):
     def __init__(self, options):
         self.expires = options.get("expires")
         self.todo = {}
-        self.futures = []
+        self.futures = {}
```

Comment:
Doesn't this (and the `done_callback` and `add` changes below) duplicate part of PR #665?

Reply:
Looks like bbb1899 moves the `done_callback` implementation from #665 up above `add`.